### PR TITLE
Core: Remove deprecated classes related to rewrite data files

### PR DESCRIPTION
**Pull request by ajantha-bhat**
These classes were kept because it was used in Spark 3.2. 
Now that the spark 3.2 is removed, we can remove these deprecated classes.

Note:
These classes were deprecated in 1.3.0. So, it can be safely removed in 1.5.0

### CODE REVIEW

File: spark/v3.3/spark/src/jmh/java/org/apache/iceberg/spark/action/IcebergSortCompactionBenchmark.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 import org.apache.iceberg.SortDirection;
 import org.apache.iceberg.SortOrder;
 import org.apache.iceberg.Table;
-import org.apache.iceberg.actions.BinPackStrategy;
```
Comment (ajantha-bhat) [inline spark/v3.3/spark/src/jmh/java/org/apache/iceberg/spark/action/IcebergSortCompactionBenchmark.java:38]: Looks like https://github.com/apache/iceberg/pull/7987 forgot to handle it for Spark 3.3. 

It has to be replaced with proper class now as the deprecated class is removed.

**Comment by ajantha-bhat** [comment]
Retriggering build due to flaky test from Flink
https://github.com/apache/iceberg/actions/runs/7208855885/job/19638664888?pr=9296

**Review by nastra — APPROVED** [review]

**Review by Fokko — APPROVED** [review]
LGTM! Thanks for taking care of this @ajantha-bhat

**Comment by nastra** [comment]
thanks @ajantha-bhat

**Pull request merged**